pandas/core/computation/expressions.py

```diff
@@ -234,13 +243,27 @@ def evaluate(op, a, b, use_numexpr: bool = True):
     b : right operand
     use_numexpr : bool, default True
         Whether to try to use numexpr.
+    inplace: bool, default False
+        Whether to do the op inplace.
+        If False, will replace
+        inplace op with the non-inplace version of it.
+        a must by an ndarray/EA in this case
     """
     op_str = _op_str_mapping[op]
+    inplace = inplace and op in _inplace_ops
+    if inplace:
+        inplace = np.result_type(a, b) == a.dtype
```

Comment:
docstring says `a` can be an EA, but id expect that to raise in np.result_type.  am i wrong there? 

come to think of it, is expect numexpr to choke on EAs in general